src/orion/storage/base.py

```
@@ -98,28 +98,57 @@ def update_trial(self, trial, where=None, **kwargs):
         """
         raise NotImplementedError()
 
-    def retrieve_result(self, trial, **kwargs):
+    def retrieve_result(self, trial, *args, **kwargs):
         """Fetch the result from a given medium (file, db, socket, etc..) for a given trial and
         insert it into the trial object
         """
         raise NotImplementedError()
 
 
 # pylint: disable=too-few-public-methods,abstract-method
-class StorageProtocol(BaseStorageProtocol, metaclass=Factory):
+class Storage(BaseStorageProtocol, metaclass=SingletonFactory):
     """Storage protocol is a generic way of allowing Orion to interface with different storage.
     MongoDB, track, cometML, MLFLow, etc...
 
     Examples
     --------
-    >>> StorageProtocol('track', uri='file://orion_test.json')
-    >>> StorageProtocol('legacy', experiment=...)
+    >>> Storage('track', uri='file://orion_test.json')
+    >>> Storage('legacy', experiment=...)
+
+    Note
+    ----
+    When retrieving an already initialized Storage object you should use `get_storage`.
+    `Storage()` should only be used for initialization purposes as `get_storage`
+    raises more granular error messages.
 
     """
 
     pass
 
 
+def get_storage():
+    """Return current storage
+
+    This is a wrapper around the Storage Singleton object to provide
+    better error message when it is used without being initialized.
+
+    Raises
+    ------
+    raises RuntimeError if the underlying storage
+    was not initialized prior to calling this function.
+
+    Note
+    ----
+    To initialize the underlying storage you must first call `Storage(...)`
+    with the appropriate arguments for the chosen backend
+
+    """
+    try:
+        return Storage()
+    except TypeError:
+        raise RuntimeError('Singleton `Storage` was not initialized!')
```

This may silence errors as well.